### PR TITLE
*: rename tidb_back_off_weight

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
The `back_off` in `tidb_back_off_weight` is not consistent with the old variable`tidb_backoff_lock_fast`, which makes the user confused.

### What is changed and how it works?
It is a new variable, may not be used widely by users, so change it to `tidb_backoff_weight`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

- Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: session/bootstrap.go [hunk 1]
```diff
@@ -853,6 +858,10 @@ func upgradeToVer34(s Session) {
 	doReentrantDDL(s, CreateOptRuleBlacklist)
 }
 
+func upgradeToVer35(s Session) {
+	mustExecute(s, "UPDATE HIGH_PRIORITY mysql.GLOBAL_VARIABLES SET VARIABLE_NAME='tidb_backoff_weight' WHERE VARIABLE_NAME='tidb_back_off_weight'")
```
Comment (zimulala) [inline session/bootstrap.go]: Do we need to update other variables? If we need to, I think we'd better do it in the same PR. Then we can reduce versions.

Comment (jackysp) [reply session/bootstrap.go]: No other name needs to be changed.

**Pull request merged**